controlplane/eks/api/v1beta2/awsmanagedcontrolplane_types.go

```diff
@@ -34,181 +34,18 @@ const (
 
 // AWSManagedControlPlaneSpec defines the desired state of an Amazon EKS Cluster.
 type AWSManagedControlPlaneSpec struct { //nolint: maligned
+	AWSManagedControlPlaneClassSpec `json:",inline"`
```

Comment:
I don't think this is necessary to change, just reference the original `AWSManagedControlPlaneSpec` in the `AWSManagedControlPlaneTemplateResource`. This would prevent backwards incompatible Go API changes that can be seen in the changes being made in the tests.

Reply:
@alexander-demicev I've pushed https://github.com/alexander-demicev/cluster-api-provider-aws/pull/2 which includes reverting the existing API changes. I have tested locally and can create EKS clusters perfectly with this API via ClusterClass.